serializers/src/main/java/io/pravega/schemaregistry/serializers/MultiFormatSerializerFactory.java

```diff
@@ -0,0 +1,199 @@
+        map.put(SerializationFormat.Protobuf, protobuf);
+
+        deserializers.forEach((key, value) -> {
+            map.put(key, new AbstractPravegaDeserializer<Object>(groupId, schemaRegistryClient, null, false, config.getDecoder(), encodingCache) {
+                @Override
+                protected Object deserialize(InputStream inputStream, SchemaInfo writerSchema, SchemaInfo readerSchema) {
+                    return value.deserialize(inputStream, writerSchema, readerSchema);
+                }
+            });
+        });
+
+        return new MultipleFormatDeserializer<>(groupId, schemaRegistryClient, map, config.getDecoder(),
+                encodingCache, transform);
+    }
+
+    private static <T> Serializer<WithSchema<T>> deserializerInternal(SerializerConfig config, Map<SerializationFormat,
+            PravegaDeserializer<Object>> deserializers, BiFunction<SerializationFormat, Object, T> transform) {
+        String groupId = config.getGroupId();
+        SchemaRegistryClient schemaRegistryClient = config.getRegistryConfigOrClient().isLeft() ?
+                SchemaRegistryClientFactory.createRegistryClient(config.getRegistryConfigOrClient().getLeft()) :
+                config.getRegistryConfigOrClient().getRight();
+        autoCreateGroup(schemaRegistryClient, config);
+        failOnCodecMismatch(schemaRegistryClient, config);
+        EncodingCache encodingCache = new EncodingCache(groupId, schemaRegistryClient);
+
+        AbstractPravegaDeserializer json = new JsonGenericDeserlizer(config.getGroupId(), schemaRegistryClient,
+                config.getDecoder(), encodingCache);
+        AbstractPravegaDeserializer protobuf = new ProtobufGenericDeserlizer(groupId, schemaRegistryClient, null, config.getDecoder(),
+                encodingCache);
+        AbstractPravegaDeserializer avro = new AvroGenericDeserlizer(groupId, schemaRegistryClient, null, config.getDecoder(),
+                encodingCache);
+
+        Map<SerializationFormat, AbstractPravegaDeserializer> map = new HashMap<>();
+        map.put(SerializationFormat.Json, json);
+        map.put(SerializationFormat.Avro, avro);
+        map.put(SerializationFormat.Protobuf, protobuf);
+
+        deserializers.forEach((key, value) -> {
+            map.put(key, new AbstractPravegaDeserializer<Object>(groupId, schemaRegistryClient, null, false, config.getDecoder(), encodingCache) {
+                @Override
+                protected Object deserialize(InputStream inputStream, SchemaInfo writerSchema, SchemaInfo readerSchema) {
+                    return value.deserialize(inputStream, writerSchema, readerSchema);
+                }
+            });
+        });
+
+        return new MultiFormatWithSchemaDeserializer<>(groupId, schemaRegistryClient, map, config.getDecoder(),
+                encodingCache, transform);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static AbstractPravegaSerializer<Object> getPravegaSerializer(
```

Comment:
Please use a generic instead of "Object"

Reply:
this is java.Object
Using `?` generic wouldnt work because the for AbstractPravegaDeserializer which is typed with generic `T`, we need to provide a type. Hence the java object.